adapters/saml/core/src/main/java/org/keycloak/adapters/saml/rotation/SamlDescriptorPublicKeyLocator.java

```diff
@@ -179,7 +180,10 @@ private synchronized <T> Key refreshCertificateCacheAndGet(T key, Map<T, Key> ca
                 this.publicKeyCacheByKey.put(new KeyHash(x509certificate.getPublicKey()), x509certificate.getPublicKey());
             } else {
                 final X500Principal principal = x509certificate.getSubjectX500Principal();
-                String name = (principal == null ? "unnamed" : principal.getName()) + "@" + x509certificate.getSerialNumber() + "$" + UUID.randomUUID();
+                SecureRandom random = new SecureRandom();
+                byte[] bytes = new byte[16];
+                random.nextBytes(bytes);
```

Comment:
Would not be better to use `SecretGenerator.randomBytes(16)`? It's in the common package so I think you can use in any other package.